src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
```

Comment:
PullProofAI

The code for the `FeedbackSecretGuard` class looks well-implemented. It checks for a feedback secret in the request headers and compares it with the expected secret from the environment variables. If they match, it allows access; otherwise, it throws an `UnauthorizedException`.

No issues found in this code snippet.


<!-- This is an auto-generated comment by PullProofAI -->